[BUG] Failed to get map preferences: SqliteError: no such column: userId

**Describe the bug**
I'm not really sure how serious the issue is, since I don't notice anything unusual. It's just that when the page loads, this appears in the log:
```
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]: [ERROR] [MiscRepository] Failed to get map preferences: SqliteError: no such column: userId
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at Database.prepare (/app/node_modules/better-sqlite3/lib/methods/wrappers.js:5:21)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLiteSession.prepareQuery (file:///app/node_modules/drizzle-orm/better-sqlite3/session.js:23:30)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLiteSession.prepareOneTimeQuery (file:///app/node_modules/drizzle-orm/sqlite-core/session.js:141:17)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLiteSession.all (file:///app/node_modules/drizzle-orm/sqlite-core/session.js:164:17)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLite3Database.all (file:///app/node_modules/drizzle-orm/sqlite-core/db.js:272:25)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at MiscRepository.executeQuery (file:///app/dist/db/repositories/base.js:111:28)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at MiscRepository.getMapPreferences (file:///app/dist/db/repositories/misc.js:984:37)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at DatabaseService.getMapPreferencesAsync (file:///app/dist/services/database.js:8233:30)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at file:///app/dist/server/server.js:4770:51
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17) {
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:   code: 'SQLITE_ERROR'
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]: }
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]: [ERROR] [MiscRepository] Failed to get map preferences: SqliteError: no such column: userId
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at Database.prepare (/app/node_modules/better-sqlite3/lib/methods/wrappers.js:5:21)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLiteSession.prepareQuery (file:///app/node_modules/drizzle-orm/better-sqlite3/session.js:23:30)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLiteSession.prepareOneTimeQuery (file:///app/node_modules/drizzle-orm/sqlite-core/session.js:141:17)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLiteSession.all (file:///app/node_modules/drizzle-orm/sqlite-core/session.js:164:17)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at BetterSQLite3Database.all (file:///app/node_modules/drizzle-orm/sqlite-core/db.js:272:25)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at MiscRepository.executeQuery (file:///app/dist/db/repositories/base.js:111:28)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at MiscRepository.getMapPreferences (file:///app/dist/db/repositories/misc.js:984:37)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at DatabaseService.getMapPreferencesAsync (file:///app/dist/services/database.js:8233:30)
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at file:///app/dist/server/server.js:4770:51
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:     at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17) {
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]:   code: 'SQLITE_ERROR'
Apr 18 10:53:53 sigaba systemd-meshmonitor[3238]: }
```

**To Reproduce**
Steps to reproduce the behavior:
1. Open MeshMonitor page
2. Perform login
3. See error

**Expected behavior**
I just don't think it should throw that error at the log level.

**Environment:**
* Host OS: Linux
* Deployment type: Docker (Podman)
  * if Docker, which Container version are you using: f5fc1152f3bb (latest) v3.12.0
* Any networking customizations: nginx reverse proxy but it has no bearing on this
* Database backend: SQLite
* Node type: Heltec V3
* Node Connection Type: Wifi

**Additional context**
The only thing I can think of is that a database migration might not have been done correctly. In any case, I’m happy to run whatever tests are necessary.

--- a/.github/workflows/system-tests.yml
+++ b/.github/workflows/system-tests.yml
@@ -14,9 +14,12 @@ on:
       - 'tests/test-*.sh'
       - '.github/workflows/system-tests.yml'
 
-# Only one system test at a time — hardware node is a shared resource
+# Scope concurrency per-PR/ref so different PRs don't cancel each other's runs.
+# The self-hosted hardware runner physically serializes execution, so runs from
+# distinct PRs will queue naturally. Only supersede in-progress runs on the SAME
+# branch (e.g. when a new commit is pushed to the same PR).
 concurrency:
-  group: system-tests
+  group: system-tests-${{ github.head_ref || github.ref }}
   cancel-in-progress: true
 
 permissions:

--- a/src/db/repositories/misc.mapprefs.test.ts
+++ b/src/db/repositories/misc.mapprefs.test.ts
@@ -1,0 +1,24 @@
+/**
+ * Regression test for issue #2713.
+ *
+ * The SQLite baseline (migration 001) creates `user_map_preferences` with
+ * `user_id` (snake_case), matching every other SQLite table's FK convention.
+ * Drizzle's SQLite schema for this table must therefore map the JS `userId`
+ * property to the physical column `user_id` — otherwise every page load that
+ * fetches map preferences logs:
+ *   SqliteError: no such column: userId
+ *
+ * (The repo's getMapPreferences catches and swallows the error, so we can't
+ * observe it by call result. Pin the schema mapping instead.)
+ */
+import { describe, it, expect } from 'vitest';
+import * as schema from '../schema/index.js';
+
+describe('userMapPreferencesSqlite — SQL column alignment (#2713)', () => {
+  it('maps JS `userId` → SQL column `user_id` to match v3.7 baseline DDL', () => {
+    const col = (schema.userMapPreferencesSqlite as unknown as {
+      userId: { name: string };
+    }).userId;
+    expect(col.name).toBe('user_id');
+  });
+});

--- a/src/db/schema/misc.ts
+++ b/src/db/schema/misc.ts
@@ -90,7 +90,7 @@ export const customThemesPostgres = pgTable('custom_themes', {
 
 export const userMapPreferencesSqlite = sqliteTable('user_map_preferences', {
   id: integer('id').primaryKey({ autoIncrement: true }),
-  userId: integer('userId').notNull().references(() => usersSqlite.id, { onDelete: 'cascade' }),
+  userId: integer('user_id').notNull().references(() => usersSqlite.id, { onDelete: 'cascade' }),
   centerLat: real('centerLat'),
   centerLng: real('centerLng'),
   zoom: real('zoom'),

--- a/src/services/database.ts
+++ b/src/services/database.ts
@@ -1439,10 +1439,16 @@ class DatabaseService {
     `);
 
     // eslint-disable-next-line no-restricted-syntax -- bootstrap: runs before migrations (Task 2.9)
+    // Column name must match baseline 001 (user_id, snake_case). Older bootstraps
+    // used camelCase `userId`, which diverged from the baseline once #2681 moved
+    // the repo to Drizzle — the Drizzle schema reads `user_id` (matching the
+    // baseline), so a legacy `userId NOT NULL` column makes inserts fail with
+    // `NOT NULL constraint failed: user_map_preferences.userId` on the first
+    // POST /api/user/map-preferences. See #2713.
     this.db.exec(`
       CREATE TABLE IF NOT EXISTS user_map_preferences (
         id INTEGER PRIMARY KEY AUTOINCREMENT,
-        userId INTEGER NOT NULL,
+        user_id INTEGER NOT NULL,
         centerLat REAL,
         centerLng REAL,
         zoom REAL,
@@ -1454,7 +1460,7 @@ class DatabaseService {
         sortDirection TEXT DEFAULT 'asc',
         createdAt INTEGER NOT NULL,
         updatedAt INTEGER NOT NULL,
-        FOREIGN KEY (userId) REFERENCES users(id) ON DELETE CASCADE
+        FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
       );
     `);
 

--- a/tests/test-api-exercise-all-backends.sh
+++ b/tests/test-api-exercise-all-backends.sh
@@ -121,6 +121,10 @@ for profile in sqlite postgres mysql; do
         echo -e "${RED}✗ $profile: API exercise test FAILED${NC}"
         RESULTS+=("$profile: FAILED")
         TOTAL_FAIL=$((TOTAL_FAIL + 1))
+        # Dump container logs so CI can diagnose the failure
+        echo -e "${YELLOW}--- $CONTAINER logs (last 200 lines) ---${NC}"
+        docker logs --tail 200 "$CONTAINER" 2>&1 || true
+        echo -e "${YELLOW}--- end $CONTAINER logs ---${NC}"
     fi
 
     # Stop backend and clean volumes

